FuelLabs/fuel-indexer
# chore: add function to get test contract ID from binary file

## Changelog
- Add function to get the contract ID of test contract from compiled binary
- Set `contract_id` field in manifest to be empty and adjust tests to fill it at test time
-- I'm okay with doing this as we've added the "make sure you change your contract ID" blurb into the docs refresh
- Remove unused testing functions
- Remove `CURRENT_TEST_CONTRACT_ID_STR` constant
- Adjust `hello-world-data` to not use a hard-coded contract ID

## Testing Plan
Adjustments were made to the E2E tests; CI should pass.

The `hello-world-data` package was adjusted to remove a hard-coded contract ID string. To ensure that it works, run the following steps:

0. Ensure your DB is cleared.
1. `cargo run --bin fuel-node -- --contract-bin examples/hello-world/contracts/greeting/out/debug/greeting.bin`
2. `cargo run --bin fuel-indexer -- run --manifest examples/hello-world/hello-index/hello_index.manifest.yaml`
3. `cargo run --bin hello-world-data -- --host 0.0.0.0:4000`
4. Ensure that the test data was successfully indexed.

## Notes
With this PR, we no longer have to adjust the contract ID fields in multiple places when making adjustments to the test contract.
